flytekit/core/type_engine.py

```diff
@@ -488,12 +488,17 @@ def get_literal_type(self, t: Type[T]) -> LiteralType:
 
         ts = TypeStructure(tag="", dataclass_type=literal_type)
 
-        return _type_models.LiteralType(simple=_type_models.SimpleType.STRUCT, metadata=schema, structure=ts)
+        return _type_models.LiteralType(simple=_type_models.SimpleType.JSON, metadata=schema, structure=ts)
 
     def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], expected: LiteralType) -> Literal:
+        import msgpack
+
+        from flytekit.models.literals import Json
+
         if isinstance(python_val, dict):
             json_str = json.dumps(python_val)
-            return Literal(scalar=Scalar(generic=_json_format.Parse(json_str, _struct.Struct())))
+            json_bytes = msgpack.dumps(json_str)
```

Comment:
do we really need to encode json objects to a string and then use msgpack to serialize the json-encoded string?

In other words, why can't we msgpack-encode `python_val` directly? As in  `msgpack.dumps(python_val)`

Reply:
In this case I think it's because the Mashumaro JSON encoder (or available to_json method) is more feature rich than `msgpack`.

That being said, this could probably use a `to_dict` -> `msgpack.dumps` and `msgpack.reads` -> `from_dict` flow for the data classes that need it (extending the mashumaro mix in), instead of `to_json`/`from_json`. The rest of them can use mashumaro's [msgpack encoder](https://github.com/Fatal1ty/mashumaro?tab=readme-ov-file#messagepack).

Reply:
yes, @JackUrb you are right, I am doing performance testing for alternatives.